src/components/toggle/Toggle.tsx

```diff
@@ -0,0 +1,104 @@
+import { useState, useLayoutEffect, useId } from 'react';
+
+import { Switch } from '@headlessui/react';
+import clsx from 'clsx';
+
+import type { Sizes } from './types';
+
+type Props = {
+  defaultEnabled?: boolean;
+  checked?: boolean;
+  disabled?: boolean;
+  label?: string;
+  className?: string;
+  size?: Sizes;
+  onChange?(checked: boolean): void;
+};
+
+const sizeStyles: { [K in Sizes]: { circle: string; background: string } } = {
+  small: {
+    circle: 'h-4 w-4',
+    background: 'h-5 w-9',
+  },
+  medium: {
+    circle: 'h-5 w-5',
+    background: 'h-6 w-11',
+  },
+  large: {
+    circle: 'h-6 w-6',
+    background: 'h-7 w-[3.25rem]',
+  },
+};
+
+const translateXConfig: { [K in Sizes]: string } = {
+  small: 'translate-x-4',
+  medium: 'translate-x-5',
+  large: 'translate-x-6',
+};
+
+const labelFontConfig: { [K in Sizes]: string } = {
+  small: 'text-sm',
+  medium: '',
+  large: 'text-xl',
+};
+
+const Toggle = ({
+  checked,
+  defaultEnabled = false,
+  label = '',
+  className = '',
+  onChange,
+  disabled = false,
+  size = 'medium',
+}: Props) => {
+  const [enabled, setEnabled] = useState(defaultEnabled);
+
+  const id = useId();
+
+  const handleOnchange = (isChecked: boolean) => {
+    setEnabled(isChecked);
+    if (onChange) onChange(isChecked);
+  };
+
+  useLayoutEffect(() => {
+    if (typeof checked === 'boolean') setEnabled(checked);
+  }, [checked]);
+
+  console.log('enabled', enabled);
+
+  return (
+    <div className="flex items-center">
+      <Switch
+        checked={enabled}
+        onChange={handleOnchange}
+        className={clsx(
+          'relative inline-flex shrink-0 cursor-pointer rounded-full border-2 border-transparent transition-colors duration-200 ease-in-out focus:outline-none focus:ring-2 focus:ring-primary-500 focus:ring-offset-2 focus:ring-offset-primary-900',
+          sizeStyles[size].background,
+          disabled ? 'cursor-not-allowed !bg-opacity-50' : '',
```

Comment:
I don't like `cursor-not-allowed`, just give it `pointer-events-none`.

And opacity seems not working.